Homework/2/Program.cs

```diff
@@ -0,0 +1,390 @@
+            Console.Clear();
+
+            int index = userArray.Length - 1;
+
+            if (removeFromStart)
+            {
+                index = 0;
+            }
+
+            if (!removeFromEnd && !removeFromStart)
+            {
+                string indexStr = PromptInput("Enter index to remove user at:");
+                index = int.Parse(indexStr);
+            }
+
+            if (index < 0 || index >= userArray.Length)
+            {
+                if (RetryConfirmation("Bad index."))
+                {
+                    PromptRemoveUser(removeFromEnd, removeFromStart);
+                }
+                else
+                {
+                    MenuSelection();
+                }
+            }
+
+            RemoveUser(index);
+            ListUsers();
+        }
+
+        //Remove user from userArray at index.
+        static void RemoveUser(int index) {
+            string[][] newArray = new string[userArray.Length - 1][];
+
+            for (int i = 0; i < newArray.Length; i++)
+            {
+                if (i <index)
+                {
+                    newArray[i] = userArray[i];
+                }
+
+                if (i == index)
+                {
+                    newArray[i] = userArray[i + 1];
+                }
+
+                if (i > index)
+                {
+                    newArray[i] = userArray[i + 1];
+                }
+            }
+
+            userArray = newArray;
+        }
+
+        //Sorting users
+        static void SortUsers() {
+            Console.Clear();
+            string order = PromptInput("Choose sort type:\n1 - asc\n2 - dec");
+            if (order != "1" && order != "2")
+            {
+                if (RetryConfirmation("Bad sort selected"))
+                {
+                    SortUsers();
+                }
+                else
+                {
+                    MenuSelection();
+                }
+            }
+
+            string[] temp;
+            for (int j = 0; j < userArray.Length; j++)
+            {
+                for (int i = 0; i < userArray.Length - 1; i++)
+                {
+                    if (order == "1")
+                    {
+                        if (userArray[i][0].CompareTo(userArray[i + 1][0]) > 0)
+                        {
+                            temp = userArray[i];
+                            userArray[i] = userArray[i + 1];
+                            userArray[i + 1] = temp;
+                        }
+                    }
+                    else {
+                        if (userArray[i][0].CompareTo(userArray[i + 1][0]) < 0)
+                        {
+                            temp = userArray[i];
+                            userArray[i] = userArray[i + 1];
+                            userArray[i + 1] = temp;
+                        }
+                    }
+                   
+                }
+            }
+            ListUsers();
+        }
```

Comment:
Please note that this part wasn't really required, but I assume you took the homework assignment and wanted to apply some context to it, which is not a bad thing.